### PR TITLE
Introduce Scope::getMaybeDefinedVariables

### PR DESCRIPTION
**Pull request by ruudk**
Fixes phpstan/phpstan#11772

Where would be the best way to create a test for this?

### CODE REVIEW

**Comment by ruudk** [comment]
Every time 🙈

**Comment by staabm** [comment]
I think a test could fit in

https://github.com/phpstan/phpstan-src/blob/1.12.x/tests/PHPStan/Analyser/ScopeTest.php

**Comment by ruudk** [comment]
@staabm Thanks. Somehow I could not find it when using `Go to test` in PHPStorm. I fixed that now.

The tests are added.

@ondrejmirtes This is ready for review!

**Comment by ondrejmirtes** [comment]
Thank you!

File: src/Analyser/MutatingScope.php [hunk 1]
```diff
@@ -574,6 +574,27 @@ public function getDefinedVariables(): array
 		return $variables;
 	}
 
+	/**
+	 * @api
+	 * @return array<int, string>
```
Comment (janedbal) [inline src/Analyser/MutatingScope.php:579]: why not `list<string>` ?

Comment (ruudk) [reply src/Analyser/MutatingScope.php:579]: Copied from getDefinedVars

Comment (janedbal) [reply src/Analyser/MutatingScope.php:579]: Fixed: https://github.com/phpstan/phpstan-src/pull/3530/files

Comment (ruudk) [reply src/Analyser/MutatingScope.php:579]: Great 🎉

**Pull request merged**